File: C_C++/Gui_thu.cpp
#include <iostream>
#include <cstring>
using namespace std;

void guiThu(string &Q, int &k)
{
	cin >> Q >> k;
	int doDai = Q.length();
	for(int i=k-1;i>=0;i--)
		cout << Q[i];
	for(int i=doDai-1;i>=k;i--)
		cout << Q[i];
}

int main()
{
	string Q;
	int k;
	guiThu(Q,k);
}

File: Python/Gui_thu.py
def guiThu(Q,k):
    Q = list(Q)
    doDai = len(Q)
    Q1 = Q[0:k]; Q1.reverse();
    Q2 = Q[k:doDai]; Q2.reverse();
    Q = Q1 + Q2
    Q = "".join(map(str,Q))
    print(Q)

Q = str(input())
k = int(input())

guiThu(Q,k)